bin/moveAndCheckSamplesheets.sh

```diff
@@ -192,41 +185,111 @@ samplesheetsSource="${DAT_ROOT_DIR}/samplesheets/new/"
 readarray -t samplesheets < <(find "${samplesheetsSource}" -maxdepth 1 -mindepth 1 -type f -name "*.${SAMPLESHEET_EXT}")
 if [[ "${#samplesheets[@]}" -eq '0' ]]
 then
+#	logTimeStamp="$(date "+%Y-%m-%d")"
+#	logDir="${DAT_ROOT_DIR}/logs/${logTimeStamp}/"
+	# shellcheck disable=SC2174
+#	mkdir -m 2770 -p "${logDir}"
+#	touch "${logDir}"
+#	export JOB_CONTROLE_FILE_BASE="${logDir}/${logTimeStamp}.${SCRIPT_NAME}"
+#	printf '' > "${JOB_CONTROLE_FILE_BASE}.started"
+	
 	log4Bash 'INFO' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "No samplesheets found in ${samplesheetsSource}."
-	mv -v "${JOB_CONTROLE_FILE_BASE}."{started,finished}
+#	mv -v "${JOB_CONTROLE_FILE_BASE}."{started,finished}
 	trap - EXIT
 	exit 0
 fi
 
 for samplesheet in "${samplesheets[@]}"
 do
+	sampleSheetName=$(basename "${samplesheet%%.*}")
+	logDir="${DAT_ROOT_DIR}/logs/${sampleSheetName}/"
+	# shellcheck disable=SC2174
+	mkdir -m 2770 -p "${logDir}"
+	touch "${logDir}"
+	export JOB_CONTROLE_FILE_BASE="${logDir}/${sampleSheetName}.${SCRIPT_NAME}"
+	printf '' > "${JOB_CONTROLE_FILE_BASE}.started"
+	
 	declare -a _sampleSheetColumnNames=()
 	declare -A _sampleSheetColumnOffsets=()
-	
+
 	IFS="${SAMPLESHEET_SEP}" read -r -a _sampleSheetColumnNames <<< "$(head -1 "${samplesheet}")"
 	
 	for (( _offset = 0 ; _offset < ${#_sampleSheetColumnNames[@]} ; _offset++ ))
 	do
 		_sampleSheetColumnOffsets["${_sampleSheetColumnNames[${_offset}]}"]="${_offset}"
 	done
+	
+	if [[ -n "${_sampleSheetColumnOffsets["SentrixBarcode_A"]+isset}" ]] 
+	then
+		log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "This is a GAP samplesheet, there is no samplesheetCheck at this moment"
+	else
+		log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "This is a NGS samplesheet, lets check if the samplesheet is correct"
+		cp "${samplesheet}"{,.converted}
+	
+		#
+		# Make sure
+		#  1. The last line ends with a line end character.
+		#  2. We have the right line end character: convert any carriage return (\r) to newline (\n).
+		#  3. We remove empty lines.
+		#
+		printf '\n'     >> "${samplesheet}.converted"
+		sed 's/\r/\n/g' "${samplesheet}.converted" > "${samplesheet}.converted.tmp"
+		sed '/^\s*$/d'  "${samplesheet}.converted.tmp" > "${samplesheet}.converted.tmp2"
+		rm "${samplesheet}.converted.tmp"
+		mv "${samplesheet}.converted.tmp2" "${samplesheet}.converted"
+		if checkSampleSheet.py --input "${samplesheet}.converted" --log "${samplesheet}.converted.log"
+		then
+			log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "samplesheet is correct"
+			check=$(cat "${samplesheet}.converted.log")
+		else
+			log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "samplesheet is not correct"
+			check=$(cat "${samplesheet}.converted.log")
+		fi
+		projectSamplesheet="false"
+		if [[ "${check}" == *'OKAY'* ]]
+		then
+			log4Bash 'INFO' "${LINENO}" "${FUNCNAME:-main}" '0' "Samplesheet is correct, continue"
```

Comment:
This looks redundant as you already concluded based on only the exit status of ```if checkSampleSheet.py ...``` whether the samplesheet was correct or not.